kliff/trainer/torch_trainer_utils/dataloaders.py

```diff
@@ -0,0 +1,173 @@
+import torch
+from torch.utils.data import Dataset as TorchDataset
+from torch_geometric.data import Dataset as TorchGeometricDataset
+
+from kliff.dataset import Dataset
+from kliff.transforms.configuration_transforms.graphs import PyGGraph
+
+
+class NeighborListDataset(TorchDataset):
+    """
+    TODO: format this dataset option
+    batch should return a tuple of (species, config, neighbor_list, neighbors, contribution_index)
+    """
+
+    def __init__(self, dataset: Dataset):
+        self.dataset = dataset
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def __getitem__(self, idx):
+        return self.dataset[idx].fingerprint
+
+
+class DescriptorDataset(TorchDataset):
+    """
+    This class is a wrapper for the :class:`torch.utils.data.Dataset` class to enable
+    the use of :class:`kliff.dataset.Dataset` as a data source for the descriptor based models.
+    It returns the fingerprint, properties and contribution index for each configuration.
+
+    Args:
+        dataset: :class:`kliff.dataset.Dataset` instance.
+        property_keys: List of property keys to be used for the training. Default is
+            ("energy", "forces", "stress"). These properties should be present on every
+            configuration in the dataset. Among the provided list of properties, any
+            property that is not present in any configuration will be ignored for all
+            configurations.
+    """
+
+    def __init__(self, dataset: Dataset, property_keys=("energy", "forces", "stress")):
+        self.dataset = dataset
+        self.dataset.check_properties_consistency(property_keys)
+        self.consistent_properties = self.dataset.get_metadata("consistent_properties")
+        self.transform = None
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def __getitem__(self, idx):
+        property_dict = {}
+        for key in self.consistent_properties:
+            property_dict[key] = getattr(self.dataset[idx], key)
+
+        if self.transform:
+            fingerprint = self.transform(self.dataset[idx], return_extended_state=True)
+        else:
+            fingerprint = self.dataset[idx].fingerprint
+        return fingerprint, property_dict
+
+    def collate(self, batch) -> dict:
+        """
+        Collate function for the dataset. This function takes in a batch of configurations
+        and properties and returns the collated configuration, properties and contribution
+        index tensors.
+        Args:
+            batch: list of configurations and properties for each configuration.
+
+        Returns:
+            dict: collated configuration, properties and contribution index tensors.
+        """
+        # get fingerprint and consistent properties
+        config_0, property_dict_0 = batch[0]
```

Comment:
No, fingerprint is the complete representation of the system, including neighbors etc. This was needed to evaluate batched gradients on descriptor. It will become more clear when I integrate the DNN trainer. Leaving this open for now.

Reply:
OK... then the name `fingerprint` is just a bit misleading. We might want to find another name for it

Reply:
How about `representation`?